lib/syntax_tree/css/selectors.rb

```diff
@@ -267,27 +337,25 @@ def relative_selector_list
 
       # <complex-selector> = <compound-selector> [ <combinator>? <compound-selector> ]*
       def complex_selector
+        consume_whitespace
+
         left = compound_selector
 
-        loop do
-          if (combinator = maybe { combinator })
-            ComplexSelector.new(left: left, combinator: combinator, right: compound_selector)
-          elsif (right = maybe { compound_selector })
-            ComplexSelector.new(left: left, combinator: nil, right: right)
-          else
-            break
-          end
+        if (c = maybe { combinator })
+          ComplexSelector.new(left: left, combinator: c, right: complex_selector)
+        elsif (right = maybe { complex_selector })
+          ComplexSelector.new(left: left, combinator: nil, right: right)
+        else
+          left
         end
-
-        left
       end
 
       # <relative-selector> = <combinator>? <complex-selector>
       def relative_selector
-        combinator = maybe { combinator }
+        c = maybe { combinator }
 
         if combinator
```

Comment:
I think I'm not understanding this line. `combinator` here is changing from a local variable read to a method call. Is that intentional?

Reply:
As originally written, the outer variable `combinator` clobbers the method `combinator` and so the variable is always `nil`.

For example:

```ruby
#! /usr/bin/env ruby

def foo
  42
end

def bar(&block)
  yield
end

foo = bar { foo }

foo
# => nil
```

I'm happy to fix it some other way, this was just the simplest I came up with at the time.

Reply:
Sorry - I missed a replacement of `combinator` with `c` on the next line, will fix it.

Reply:
Pushed a complete changeset, let me know if it needs explanation.